packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -60,10 +59,13 @@ class ProtocolDiscovery {
 
   void _handleLine(String line) {
     Uri uri;
-    final int index = line.indexOf(_prefix + 'http://');
-    if (index >= 0) {
+
+    final RegExp r = new RegExp('${RegExp.escape(serviceName)} listening on (http://[^ \n]+)');
+    final Match match = r.firstMatch(line);
+
+    if (match != null) {
       try {
-        uri = Uri.parse(line.substring(index + _prefix.length));
+        uri = Uri.parse(match[2]);
```

Comment:
This needs to be `match[1]` again now 😄

Reply:
argh, that's true, I should stop code in the iPad, fix and commit